Automattic/wp-calypso
# New Onboarding: Fix eCommerce plan not being recommended

#### Changes proposed in this Pull Request
* Fix getting a recommended plan (eCommerce) when selecting "Store" feature.
* Fix plan pre-selection in Focused launch for monthly plans.

#### Technical Changes
* `TIMELESS_*` slugs are now all lowercase. The reported bug is because the [locally defined slug](https://github.com/Automattic/wp-calypso/pull/48790/files#diff-c442fdac5b918460aacdcf99f45d7e5306f96c8628dd538905f789ea6fcc8c7dR21) is not matching `nonlocalized_short_name` for eCommerce.
* `plansSlugs` is the list of all supported plans defined in `TIMELESS_*` slugs
* Remove the hardcoded `plansPaths` list
* `isPlanProduct` utility function from launch package is now using the complete list of `plansProductSlugs`

#### Testing instructions
* Go to `/new` and advance through the flow selecting Store feature => eCommerce plan should be recommended
* Create a site on `/start` and choose a monthly plan and leave it in cart => open Focused launch and plan should be pre-selected in Summary View (Demo: https://cloudup.com/c1ouW7Yp5le). Caveat: prices won't be correct before #49257 gets fixed.

Fixes #49357
Fixes #49452
